x-pack/plugins/security_solution/server/endpoint/lib/artifacts/lists.test.ts

```diff
@@ -513,6 +513,358 @@ describe('artifacts lists', () => {
+        mockExceptionClient.findExceptionListItem = jest.fn().mockReturnValueOnce(first);
+
+        const resp = await getFilteredEndpointExceptionList({
+          elClient: mockExceptionClient,
+          schemaVersion: 'v1',
+          filter: `${getOsFilter(os)} and (exception-list-agnostic.attributes.tags:"policy:all")`,
+          listId: ENDPOINT_TRUSTED_APPS_LIST_ID,
+        });
+        expect(resp).toEqual({
+          entries: [expectedEndpointExceptions],
+        });
+      });
+    });
+
+    describe('macos/windows', () => {
+      test('it should add process.name entry for process.executable entry with wildcard type', async () => {
+        const os = Math.floor(Math.random() * 2) === 0 ? 'windows' : 'macos';
+        const value = os === 'windows' ? 'C:\\My Doc*\\doc.md' : '/usr/bi*/doc.md';
+
+        const testEntries: EntriesArray = [
+          {
+            field: 'process.executable.caseless',
+            operator: 'included',
+            type: 'wildcard',
+            value,
+          },
+        ];
+
+        const expectedEndpointExceptions = {
+          type: 'simple',
+          entries: [
+            {
+              field: 'process.executable',
+              operator: 'included',
+              type: 'wildcard_caseless',
+              value,
+            },
+            {
+              field: 'process.name',
+              operator: 'included',
+              type: 'exact_caseless',
+              value: 'doc.md',
+            },
+          ],
+        };
+
+        const first = getFoundExceptionListItemSchemaMock();
+        first.data[0].entries = testEntries;
+        first.data[0].os_types = [os];
+        mockExceptionClient.findExceptionListItem = jest.fn().mockReturnValueOnce(first);
+
+        const resp = await getFilteredEndpointExceptionList({
+          elClient: mockExceptionClient,
+          schemaVersion: 'v1',
+          filter: `${getOsFilter(os)} and (exception-list-agnostic.attributes.tags:"policy:all")`,
+          listId: ENDPOINT_TRUSTED_APPS_LIST_ID,
+        });
+        expect(resp).toEqual({
+          entries: [expectedEndpointExceptions],
+        });
+      });
+
+      test('it should add process.name entry when wildcard file.path.text entry has filename', async () => {
+        const os = Math.floor(Math.random() * 2) === 0 ? 'windows' : 'macos';
+        const value = os === 'windows' ? 'C:\\My Doc*\\doc.md' : '/usr/bi*/doc.md';
+
+        const testEntries: EntriesArray = [
+          {
+            field: 'file.path.text',
+            operator: 'included',
+            type: 'wildcard',
+            value,
+          },
+        ];
+
+        const expectedEndpointExceptions = {
+          type: 'simple',
+          entries: [
+            {
+              field: 'file.path.text',
+              operator: 'included',
+              type: 'wildcard_caseless',
+              value,
+            },
+            {
+              field: 'process.name',
```

Comment:
@ashokaditya when we are using `file.path.text` we cannot use `process.name` as it will almost always create an invalid rule since the `process.name` in the doc will be different.

Instead, we can add `file.name` when using the `file.path.text` field.  This will create a valid rule similarly as when we use `process.executable` and `process.name` together.

Reply:
done 5bfd726acd459eacba783e356d08844c73a06fd2